src/diffusers/models/modeling_utils.py

```diff
@@ -497,143 +500,112 @@ def from_pretrained(cls, pretrained_model_name_or_path: Optional[Union[str, os.P
         You should probably TRAIN this model on a down-stream task to be able to use it for predictions and inference.
         ```
         """
-        cache_dir = kwargs.pop("cache_dir", None)
-        ignore_mismatched_sizes = kwargs.pop("ignore_mismatched_sizes", False)
-        force_download = kwargs.pop("force_download", False)
-        from_flax = kwargs.pop("from_flax", False)
-        resume_download = kwargs.pop("resume_download", False)
-        proxies = kwargs.pop("proxies", None)
-        output_loading_info = kwargs.pop("output_loading_info", False)
-        local_files_only = kwargs.pop("local_files_only", None)
-        token = kwargs.pop("token", None)
-        revision = kwargs.pop("revision", None)
-        torch_dtype = kwargs.pop("torch_dtype", None)
-        subfolder = kwargs.pop("subfolder", None)
-        device_map = kwargs.pop("device_map", None)
-        max_memory = kwargs.pop("max_memory", None)
-        offload_folder = kwargs.pop("offload_folder", None)
-        offload_state_dict = kwargs.pop("offload_state_dict", False)
-        low_cpu_mem_usage = kwargs.pop("low_cpu_mem_usage", _LOW_CPU_MEM_USAGE_DEFAULT)
-        variant = kwargs.pop("variant", None)
-        use_safetensors = kwargs.pop("use_safetensors", None)
-
-        allow_pickle = False
-        if use_safetensors is None:
-            use_safetensors = True
-            allow_pickle = True
-
-        if low_cpu_mem_usage and not is_accelerate_available():
-            low_cpu_mem_usage = False
-            logger.warning(
-                "Cannot initialize model with low cpu memory usage because `accelerate` was not found in the"
-                " environment. Defaulting to `low_cpu_mem_usage=False`. It is strongly recommended to install"
-                " `accelerate` for faster and less memory-intense model loading. You can do so with: \n```\npip"
-                " install accelerate\n```\n."
-            )
+        single_file_ckpt = False
+        if is_single_file_checkpoint(pretrained_model_name_or_path):
+            if cls.__name__ not in SINGLE_FILE_LOADABLE_CLASSES:
+                raise ValueError(
+                    f"{cls.__name__} is not supported. Supported classes are: {' '.join(list(SINGLE_FILE_LOADABLE_CLASSES))}."
+                )
+            logger.info("Single file checkpoint detected...")
+            model = cls.from_single_file(pretrained_model_name_or_path, **kwargs)
+            single_file_ckpt = True
 
-        if device_map is not None and not is_accelerate_available():
-            raise NotImplementedError(
-                "Loading and dispatching requires `accelerate`. Please make sure to install accelerate or set"
-                " `device_map=None`. You can install accelerate with `pip install accelerate`."
-            )
+        else:
```

Comment:
Instead of nesting all the `from_pretrained` logic under an `else`, why not just return the model directly if the `is_single_file_checkpoint(pretrained_model_name_or_path)` is True. e.g

```python
if is_single_file_checkpoint(pretrained_model_name_or_path):
   model = cls.from_single_file(...)
   model.eval()
   return model

## `from_pretrained` logic remains unchanged. 
```

Reply:
Agreed. Changes are in [d7d757a](https://github.com/huggingface/diffusers/pull/6986/commits/d7d757a38791c9737ba953a002121d455dcf0db4).